core-common/src/main/java/org/mapstruct/BuilderMapping.java

```diff
@@ -0,0 +1,38 @@
+/**
+ *  Copyright 2012-2017 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * @author Filip Hrisafov
+ * @since 1.3
+ */
+@Retention(RetentionPolicy.CLASS)
+@Target({
+    ElementType.TYPE,
+    ElementType.METHOD
+})
+public @interface BuilderMapping {
+
+    String builderFinisher() default "build";
```

Comment:
I thought the problem was 2 fold: `builderFinisher` and `builderCreator`.. where's the latter one?

Reply:
The `builderCreator` one is a more complex one. We need the type of the builder at the time of the creation of the `Type` itself. As we need the properties really early on, when we are constructing the `Target` and `Source` references. That's why that is not hear, and I am having hard time pulling that here.

Reply:
ok.. but then leave it open as a new issue / improvement.. From a user perspective, its strange, agreed?

Reply:
I'll create an issue for that. yes it might be a bit strange from a user perspective.

Reply:
Issue created https://github.com/mapstruct/mapstruct/issues/1547